consumers/client-event-consumer/src/main/java/com/flipkart/aesop/event/AbstractEventFactory.java

```diff
@@ -118,4 +100,40 @@ public AbstractEvent createEvent(Map<String, Object> fieldsMap, Set<String> prim
 
 		return event;
 	}
+
+	/**
+	 * This returns mysql object from avro object and intented mysql datatype
+	 * @param mysqlType
+	 * @param fieldValue
+	 * @return MysqlTypedObject using AvroToMysqlMapper
+	 */
+	private Object getMysqlTypedObject(String mysqlType, Object fieldValue)
+	{
+		MysqlDataTypes mysqlTypeToConvert = MysqlDataTypes.valueOf(mysqlType.toUpperCase());
+		Object mysqlTypedObject = AvroToMysqlMapper.avroToMysqlType(fieldValue, mysqlTypeToConvert);
+		return mysqlTypedObject;
+	}
+
+	/**
+	 * This specifically handles rowChangeField which comes in form HashMap and converting each key/value to Mysql type
+	 * @param fieldValue
+	 * @param fieldToMysqlDataType
+	 * @return MysqlTypedObject using AvroToMysqlMapper
+	 */
+	private Map<String, Object> getMysqlObjectForRowChangeField(HashMap<Object, Object> fieldValue,
+												   Map <String, String> fieldToMysqlDataType)
+	{
+		Map<String, Object> mysqlTypedObject = null;
+		if (fieldValue != null)
+		{
+			mysqlTypedObject = new HashMap<String, Object>();
```

Comment:
```java
// If possible, better to create collections with size to avoid dynamic allocation later
mysqlTypedObject = new HashMap<String, Object>(fieldValue.size()); 
for(Map.Entry<String, Object> entry : fieldValue.entrySet()) {
  String fieldName = entry.getKey();
  String sqlType = fieldToMysqlDataType.get(fieldName);
  mysqlTypedObject.put(fieldName, getMysqlTypedObject(sqlType, entry.getValue());
}
```